api/v1/storage-contract.go

```diff
@@ -34,22 +42,56 @@ type AddressedStorageContract struct {
 //	@Description <br/> *Note* CID is optional, if specified, then `storage_contract` becomes optional.
 //	@Description If both are specified, then we will validate that the CID matches the proposed storage contract and return an error if not.
 //	@Description If only CID is specified, then we will fetch it and update the storage contract to it.
-//	@Security apiKey
+//	@Security		apiKey
 //	@Param 			collection body AddressedStorageContract true "New Storage Contract to update to"
 //	@Produce		json
 //	@Success		200	{object}	ResponseEnvelope{response=AddressedStorageContract}
 //	@Router			/storage-contract [post]
-func handleSetStorageContract(c echo.Context) error {
-	return c.JSON(http.StatusNotImplemented, map[string]string{})
+func (a *apiV1) handleSetStorageContract(c echo.Context) error {
+	var addressedStorageContract AddressedStorageContract
+	err := json.NewDecoder(c.Request().Body).Decode(&addressedStorageContract)
+
+	if err != nil {
+		return c.JSON(http.StatusInternalServerError, CreateErrorResponseEnvelope(c, http.StatusInternalServerError, err.Error()))
+	}
+
+	if len(addressedStorageContract.Cid) == 0 {
+		return c.JSON(http.StatusNotImplemented, CreateErrorResponseEnvelope(c, http.StatusNotImplemented, "StorageContract is not currently supported. Please pass in a CID."))
+	}
+
+	var tenant db.Tenant
+	tenant.TenantID = db.ID(GetTenantContext(c).TenantID)
+	err = a.db.Find(&tenant).Error
+
+	if err != nil {
+		return c.JSON(http.StatusInternalServerError, CreateErrorResponseEnvelope(c, http.StatusInternalServerError, err.Error()))
```

Comment:
this will be a common thing we check - I think we should define a specific error code for it i.e, `errorFetchingTenant`

Reply:
I have a todo to come up with error codes. We need to get together and flush this out.

Reply:
I believe this also return empty and not error if the tenant isnt found